### PR TITLE
[RAPTOR-7659] make java_dropin_env_base use openjdk base image

### PR DESCRIPTION
**Pull request by yakov-g**
- created a new image in dockerhub datarobot/java-dropin-env-base:debian10-py3.7-java11
- make java_codegen env to use this image
- haven't done any optimizations to reduce image size (though it's 70 Mb less, than the previous base image)

# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

### CODE REVIEW

File: docker/java_dropin_env_base/Dockerfile [hunk 1]
```diff
@@ -1,42 +1,13 @@
 # This is the default base image for use with user models and workflows.
-FROM ubuntu:18.04
-ENV LC_ALL=en_US.UTF-8 TERM=xterm COLS=132 ROWS=43 DEBIAN_FRONTEND=noninteractive
+FROM openjdk:11.0.12-jre-slim-buster
```
Comment (zohar-mizrahi) [inline docker/java_dropin_env_base/Dockerfile:2]: Which OS is it based on?

File: docker/java_dropin_env_base/Dockerfile [hunk 2]
```diff
@@ -1,42 +1,13 @@
...
+    pip3 install --upgrade pip && \
+    pip3 install --no-cache-dir wheel setuptools && \
+    # required for mlpiper
+    apt install -y python3-setuptools && \
+    chmod 707 /var/lib/nginx
 
 # do the trick to install drum deps
-RUN pip3 install --no-cache-dir datarobot-drum
-RUN pip3 uninstall -y datarobot-drum
+RUN pip3 install --no-cache-dir datarobot-drum && pip3 uninstall -y datarobot-drum
```
Comment (zohar-mizrahi) [inline docker/java_dropin_env_base/Dockerfile:13]: Don't we want to control which `datarobot-drum` version to install?

File: public_dropin_environments/java_codegen/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # This is the default base image for use with user models and workflows.
-FROM datarobot/java-dropin-env-base
+FROM datarobot/java-dropin-env-base:debian10-py3.7-java11
```
Comment (zohar-mizrahi) [inline public_dropin_environments/java_codegen/Dockerfile:2]: Is it really debian OS? Have we tested it and certified it? Is it documented anywhere?

File: docker/java_dropin_env_base/Dockerfile (hunk 1 above)
Comment (yakov-g) [reply docker/java_dropin_env_base/Dockerfile:2]: `buster` is name for Debian 10

File: docker/java_dropin_env_base/Dockerfile (hunk 2 above)
Comment (yakov-g) [reply docker/java_dropin_env_base/Dockerfile:13]: As this is base, I just install/uninstall DRUM to pull dependencies in order to potentially reduce build time when this base is used for actual envs.

Comment (zohar-mizrahi) [reply docker/java_dropin_env_base/Dockerfile:13]: Um ... but dependencies may be different depending on the drum version

File: public_dropin_environments/java_codegen/Dockerfile (hunk 3 above)
Comment (yakov-g) [reply public_dropin_environments/java_codegen/Dockerfile:2]: We create this base so we know what is used. This is Debian

File: docker/java_dropin_env_base/Dockerfile (hunk 2 above)
Comment (yakov-g) [reply docker/java_dropin_env_base/Dockerfile:13]: Yes. But then we actually install the latest DRUM in actual env. So if versions have been changed, they will be updated.

File: public_dropin_environments/java_codegen/Dockerfile (hunk 3 above)
Comment (zohar-mizrahi) [reply public_dropin_environments/java_codegen/Dockerfile:2]: Have you ever tested Debian? I've never had experience with it. How stable it is, etc.

Comment (yakov-g) [reply public_dropin_environments/java_codegen/Dockerfile:2]: I haven't explicitly tested it. But we use it for legacy package images.
Quick google search says debian is more stable than ubuntu.

**Comment by yakov-g** [comment]
So...
Our `h2o pojo` models have to be compiled, so base image should be based not on 
openjdk:11.0.12-jre-slim-buster

but on
openjdk:11.0.12-jdk-slim-buster

**Review by zohar-mizrahi — APPROVED** [review]

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Review by jmbannon — APPROVED** [review]

**Comment by yakov-g** [comment]
> Talking to @4dahalibut , this is inference-only code. Please update codeowners. Removing the Core Modeling tag
Tag is assigned for all the public environments. Java is not used in training models. Do you want me to remove tag only for java env?
/public_dropin_environments                                         @datarobot/custom-models @datarobot/core-modeling

**Pull request merged**